AutomatticTracks/build.gradle

```diff
@@ -4,37 +4,48 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.3.1'
+        classpath 'com.android.tools.build:gradle:4.1.2'
         classpath 'com.novoda:bintray-release:0.9'
+        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.30'
     }
 }
 
 apply plugin: 'com.android.library'
 apply plugin: 'maven'
 apply plugin: 'com.novoda.bintray-release'
+apply plugin: 'kotlin-android'
 
 repositories {
     google()
     jcenter()
 }
 
 dependencies {
-    implementation 'io.sentry:sentry-android:1.7.16'
-    implementation 'com.android.support:support-annotations:28.0.0'
+    implementation 'io.sentry:sentry-android:4.1.0'
+    implementation "androidx.annotation:annotation:1.1.0"
+    testImplementation "junit:junit:4.13"
+    testImplementation "org.mockito:mockito-core:3.3.3"
+    testImplementation 'org.robolectric:robolectric:4.4'
+    testImplementation 'androidx.test:core:1.3.0'
+}
+
+project.ext {
+    versionName = "1.2.1"
 }
 
 android {
-    compileSdkVersion 26
+    compileSdkVersion 30
 
     defaultConfig {
-        versionName "1.2.1"
+        versionName "${versionName}"
         minSdkVersion 14
-        targetSdkVersion 25
+
+        buildConfigField "String", "VERSION_NAME", "${versionName}"
```

Comment:
[From AGP release notes](https://developer.android.com/studio/releases/gradle-plugin#version_properties_removed_from_buildconfig_class_in_library_projects):
> Version properties removed from BuildConfig class in library projects
For library projects only, the BuildConfig.VERSION_NAME and BuildConfig.VERSION_CODE properties have been removed from the generated BuildConfig class because these static values did not reflect the final values of the application’s version code and name, and were therefore misleading. Additionally, these values were discarded during manifest merging.

I think it's something we should handle later, I don't know how yet - probably when migrating of JCenter?

I've decided to artificially add keep previous implementation.

Reply:
I think we could retrieve this property dynamically from the package info. I have seen some stackoverflow threads, but I haven't had a chance to try it yet. Maybe it'd be worth looking into as we may miss this in future refactors. At the very least, I think we should add a TODO for it.

I agree with their decision to remove it, because a lot of the time, the version name will not actually match the value in the build file. This is because publishing a new version would require the build file to change and that's not fun to do in the CI or ask developers to make sure of. It's just easier (better?) to rely on information from git. (such as tags)

Reply:
In #54 we bumped AGP to `4.0.2` instead, so this field is still available. Let's go back to this when updating AGP to `4.1.x`